test/integration/suite_test.go

```diff
@@ -50,6 +50,20 @@ func TestMain(m *testing.M) {
 			WithProxyAdminServiceTypeLoadBalancer()
 	}
 
+	if kongImage != "" {
+		if kongTag == "" {
+			exitOnErrWithCode(fmt.Errorf("TEST_KONG_IMAGE requires TEST_KONG_TAG"), ExitCodeEnvSetupFailed)
+		}
+		kongbuilder = kongbuilder.WithProxyImage(kongImage, kongTag)
+	}
+
+	if kongPullUsername != "" {
+		if kongPullPassword == "" {
+			exitOnErrWithCode(fmt.Errorf("TEST_KONG_PULL_USERNAME requires TEST_KONG_PULL_PASSWORD"), ExitCodeEnvSetupFailed)
+		}
+		kongbuilder = kongbuilder.WithProxyImagePullSecret("", kongPullUsername, kongPullPassword, "")
+	}
```

Comment:
technically you're not catching the inverse, where a password or a tag is provided but not the other. You could solve this by doing:

```golang
if (cond1 || cond2) && !(cond1 && cond2) {
    exitOnErrWithCode(fmt.Errorf("username AND password must be provided"))
```